linera-base/src/identifiers.rs

```diff
@@ -990,6 +991,7 @@ impl Display for AccountOwner {
         match self {
             AccountOwner::User(owner) => write!(f, "User:{}", owner)?,
             AccountOwner::Application(app_id) => write!(f, "Application:{}", app_id)?,
+            AccountOwner::Chain => write!(f, "Chain")?,
```

Comment:
I look forward to remove these tags

Reply:
The `::Chain` variant might not be so easy to remove b/c there are flow paths where the `Chain` is still a valid identifier. Right now I can see two ways to get rid of it:
1. Do not let chain own a balance.
2. Turn `::Chain` variant into a special case of address (`[u8; 32]`).

The first one isn't just a refactor (internal change) as it requires making some high-level design decisions around opening new chains etc.

I think the second option is the easier one and we should try to do it. We could hash a string like `"Linera Chain Address"` (note it cannot simply be `[0u8; 32]` b/c in Ed25519 a private key for _all zeros public key_ is [well-known](https://substrate.stackexchange.com/questions/982/why-does-the-all-0-public-key-have-a-known-private-key-in-sr25519-and-ed25519)).

Reply:
> note it cannot simply be [0u8; 32] b/c in Ed25519 a private key for all zeros public key is [well-known](https://substrate.stackexchange.com/questions/982/why-does-the-all-0-public-key-have-a-known-private-key-in-sr25519-and-ed25519))

Addresses are not public keys so `[0u8; 32]` would work. I proposed `0x0` (`Reserved(0)`) in the internal doc.